google/cloud/bigtable/data/_sync_autogen/_mutate_rows.py

```diff
@@ -0,0 +1,182 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file is automatically generated by CrossSync. Do not edit manually.
+
+from __future__ import annotations
+from typing import Sequence, TYPE_CHECKING
+import functools
+from google.api_core import exceptions as core_exceptions
+from google.api_core import retry as retries
+import google.cloud.bigtable.data.exceptions as bt_exceptions
+from google.cloud.bigtable.data._helpers import _attempt_timeout_generator
+from google.cloud.bigtable.data._helpers import _retry_exception_factory
+from google.cloud.bigtable.data.mutations import _MUTATE_ROWS_REQUEST_MUTATION_LIMIT
+from google.cloud.bigtable.data.mutations import _EntryWithProto
+from google.cloud.bigtable.data._cross_sync import CrossSync
+
+if TYPE_CHECKING:
+    from google.cloud.bigtable.data.mutations import RowMutationEntry
+    from google.cloud.bigtable_v2.services.bigtable.client import (
+        BigtableClient as GapicClientType,
+    )
+    from google.cloud.bigtable.data._sync_autogen.client import Table as TableType
+
+
+class _MutateRowsOperation:
+    """
+    MutateRowsOperation manages the logic of sending a set of row mutations,
+    and retrying on failed entries. It manages this using the _run_attempt
+    function, which attempts to mutate all outstanding entries, and raises
+    _MutateRowsIncomplete if any retryable errors are encountered.
+
+    Errors are exposed as a MutationsExceptionGroup, which contains a list of
+    exceptions organized by the related failed mutation entries.
+
+    Args:
+        gapic_client: the client to use for the mutate_rows call
+        table: the table associated with the request
+        mutation_entries: a list of RowMutationEntry objects to send to the server
+        operation_timeout: the timeout to use for the entire operation, in seconds.
+        attempt_timeout: the timeout to use for each mutate_rows attempt, in seconds.
+            If not specified, the request will run until operation_timeout is reached.
+    """
+
+    def __init__(
+        self,
+        gapic_client: GapicClientType,
+        table: TableType,
+        mutation_entries: list["RowMutationEntry"],
+        operation_timeout: float,
+        attempt_timeout: float | None,
+        retryable_exceptions: Sequence[type[Exception]] = (),
+    ):
+        total_mutations = sum((len(entry.mutations) for entry in mutation_entries))
+        if total_mutations > _MUTATE_ROWS_REQUEST_MUTATION_LIMIT:
+            raise ValueError(
+                f"mutate_rows requests can contain at most {_MUTATE_ROWS_REQUEST_MUTATION_LIMIT} mutations across all entries. Found {total_mutations}."
+            )
+        self._gapic_fn = functools.partial(
+            gapic_client.mutate_rows,
+            table_name=table.table_name,
+            app_profile_id=table.app_profile_id,
+            retry=None,
```

Comment:
This would be the gapic-level retry logic. We set it to None because we handle retries at a higher-level, retrying `self._run_attempt` instead (It's wrapped by `CrossSync._Sync_Impl.retry_target` into `self._operation` a few lines down)